src/conduits/RwaOutputConduit3.t.sol

```diff
@@ -0,0 +1,533 @@
+
+        outputConduit.hope(address(0));
+
+        assertEq(outputConduit.can(address(0)), 1);
+
+        vm.expectEmit(true, false, false, false);
+        emit Nope(address(0));
+
+        outputConduit.nope(address(0));
+
+        assertEq(outputConduit.can(address(0)), 0);
+    }
+
+    function testKissDiss() public {
+        assertEq(outputConduit.bud(address(0)), 0);
+
+        vm.expectEmit(true, false, false, false);
+        emit Kiss(address(0));
+
+        outputConduit.kiss(address(0));
+
+        assertEq(outputConduit.bud(address(0)), 1);
+
+        vm.expectEmit(true, false, false, false);
+        emit Diss(address(0));
+
+        outputConduit.diss(address(0));
+
+        assertEq(outputConduit.bud(address(0)), 0);
+    }
+
+    function testFile() public {
+        assertEq(outputConduit.quitTo(), address(testUrn));
+
+        address quitToAddress = vm.addr(1);
+        vm.expectEmit(true, true, false, false);
+        emit File(bytes32("quitTo"), quitToAddress);
+
+        outputConduit.file(bytes32("quitTo"), quitToAddress);
+
+        assertEq(outputConduit.quitTo(), quitToAddress);
+    }
+
+    function testRevertOnFileUnrecognisedParam() public {
+        vm.expectRevert("RwaOutputConduit3/unrecognised-param");
+        outputConduit.file(bytes32("random"), address(0));
+    }
+
+    function testRevertOnUnauthorizedMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.rely(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.deny(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.hope(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.nope(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.hate(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.mate(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.kiss(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.diss(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.file(bytes32("quitTo"), address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.yank(address(0), me, 0);
+    }
+
+    function testRevertOnNotMateMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-mate");
+        outputConduit.push();
+
+        vm.expectRevert("RwaOutputConduit3/not-mate");
+        outputConduit.quit();
+    }
+
+    function testRevertOnNotOperatorMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-operator");
+        outputConduit.pick(address(0));
+    }
+
```

Comment:
I think we are missing a test for picking address(0), i.e the scenario for invalidating `to`.

Reply:
I think it's not possible to pick a zero address, `to` address can be set to a zero address only through push`.

Reply:
see here - https://github.com/makerdao/mip21-toolkit/blob/READONLY-add-mip65-components/src/conduits/RwaOutputConduit3.sol#L271

Reply:
true

Reply:
done

Reply:
I still don't think this is tested.
I see `testRevertOnPickZeroAddress` was added, but it's basically then same as `testRevertOnPickAddressNotWhitelisted`.

Reply:
Fixed